<a href="https://colab.research.google.com/github/theasad-1055/Regression_project/blob/main/simple_linear.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project **Simple LinearRegression** based

In [ ]:
# import library
import pandas as pd

In [ ]:
#import dataset
df = pd.read_csv('/placement.csv')

In [ ]:
df.head(5)

cgpa  package
0  6.89     3.26
1  5.12     1.98
2  7.82     3.25
3  7.42     3.67
4  6.94     3.57

In [ ]:
# now I create html analysi file to look data if good or bad
# first install  library
!pip install ydata-profiling


In [ ]:
from ydata_profiling import ProfileReport
import ydata_profiling
print(ydata_profiling.__version__)  # find version


v4.17.0


In [ ]:
prof = ProfileReport(df)
prof.to_file(output_file="output_p.html")  # after this execution 'output_p.html' file created in system

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]


100%|██████████| 2/2 [00:00<00:00, 70.26it/s]


Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

EDA (Explotary data analysis) perform manualy

In [9]:
df.shape

(200, 2)

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   cgpa     200 non-null    float64
 1   package  200 non-null    float64
dtypes: float64(2)
memory usage: 3.3 KB


In [12]:
df.describe()

cgpa     package
count  200.000000  200.000000
mean     6.990500    2.996050
std      1.069409    0.691644
min      4.260000    1.370000
25%      6.190000    2.487500
50%      6.965000    2.995000
75%      7.737500    3.492500
max      9.580000    4.620000

In [13]:
df.isnull().sum()

cgpa       0
package    0
dtype: int64

In [14]:
df.nunique()

cgpa       163
package    138
dtype: int64

### now create independent variable (x) and  dependent variable (y)

In [15]:
x = df.iloc[:,0:1]
y = df.iloc[:,1]

In [18]:
x.head()

cgpa
0  6.89
1  5.12
2  7.82
3  7.42
4  6.94

In [19]:
y.head()

0    3.26
1    1.98
2    3.25
3    3.67
4    3.57
Name: package, dtype: float64

### train test split data

In [20]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.2, random_state=42)

## create model to predict based on data

In [21]:
from sklearn.linear_model import LinearRegression
model = LinearRegression()

In [22]:
model.fit(x_train,y_train)

LinearRegression()

### check model prediction

In [25]:
y_pred = model.predict(x_test)

print("prediction: ", y_pred)



prediction:  [2.78031348 3.13635249 3.1995207  2.38981908 3.52684689 3.76803461
 3.16506531 2.54486832 3.17655044 3.4923915  1.90744364 2.34962112
 3.6876387  2.75734322 3.47516381 3.04447145 2.32665086 3.20526327
 2.17734418 3.314372   2.45298729 2.90090734 3.32011456 2.87219451
 3.33734226 2.19457187 1.41932564 2.7114027  3.18229301 2.32665086
 3.74506435 2.95833298 3.68189614 2.97556068 2.59080884 3.34882738
 2.47595755 3.07318428 4.17575671 2.95833298]


In [30]:
model.predict(x_test.iloc[0].values.reshape(-1,1))

/usr/local/lib/python3.12/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


array([2.78031348])

##Project Summary:
In this project, I built a simple linear regression model using a small dataset of around 200 records with no missing values. The goal was to predict a continuous target variable. The model was trained successfully, but the prediction accuracy was not very high. For example, when the actual target value was 3.26, the model predicted around 2.78. This shows that while the model is working, it can be improved with more data or better feature selection.